src/nvim/drawscreen.c

```diff
@@ -715,14 +716,17 @@ void end_search_hl(void)
   screen_search_hl.rm.regprog = NULL;
 }
 
-static void win_redr_bordertext(win_T *wp, VirtText vt, int col)
+static void win_redr_bordertext(win_T *wp, VirtText vt, int col, BorderTextType bt)
 {
   for (size_t i = 0; i < kv_size(vt);) {
-    int attr = 0;
+    int attr = -1;
     char *text = next_virt_text_chunk(vt, &i, &attr);
     if (text == NULL) {
       break;
     }
+    if (attr == -1) {  // No highlight specified.
+      attr = bt == kBorderTextTitle ? wp->w_ns_hl_attr[HLF_BTITLE] : wp->w_ns_hl_attr[HLF_BFOOTER];
```

Comment:
can reduce the scope of the ternary to `wp->w_ns_hl_attr[bt == kBorderTextTitle ? HLF_BTITLE : HLF_BFOOTER]